internal/datacoord/services.go

```diff
@@ -81,6 +81,18 @@ func (s *Server) Flush(ctx context.Context, req *datapb.FlushRequest) (*datapb.F
 		}, nil
 	}
 
+	channelCPs := make(map[string]*msgpb.MsgPosition, 0)
+	coll, exist := s.meta.collections[req.GetCollectionID()]
```

Comment:
Please don't access the meta collections directly(without lock guard), use `s.meta.GetCollection()` instead.
By the way, you may get a nill collection here before this collection is cached in datacoord, use `handler.GetCollection()` can avoid that.

Reply:
OK